cypress/component/DataSearch/dataset_search_table.spec.js

```diff
@@ -38,6 +42,56 @@ describe('Dataset Search Table tests', () => {
       cy.get('#header-checkbox').click();
       cy.contains('1 dataset selected from 1 study');
     });
+  });
+
+
+  describe('Data library filter by participant count tests', () => {
+
+    beforeEach(() => {
+      cy.initApplicationConfig();
+      cy.stub(TerraDataRepo, 'listSnapshotsByDatasetIds').returns({});
+      cy.clock();
+    });
+
+    function handler(request, searchText) {
+      if (JSON.stringify(request.body).includes(searchText)) {
+        request.reply(['filtered']);
+      } else {
+        request.reply([]);
+      }
+    }
+
+
+    it('When a participant count filter is applied the query is updated', () => {
+      cy.intercept(
+        {method: 'POST', url: '**/search/index'}, (req) => {
+          return handler(req, '{"range":{"participantCount":{"gte":null,"lte":50}}}');
+        }).as('searchIndex');
+      mount(<DatasetSearchTable {...props} />);
+      // first clear the default value (100), without clearing first, type('50') would result in input of 10050
+      cy.get('#participantCountMax-range-input').clear().type('50');
+      cy.tick(150);
+      // this api call should have had a request that contained the searchText
+      cy.wait('@searchIndex').then((response) => {
+        expect(response.response.body[0]).to.equal('filtered');
+      });
+      cy.get('@searchIndex.all').should('have.length', 1);
```

Comment:
I can see that this test does pass, but I'm confused as to what the `.all` part of this alias selector is doing and where it is defined.

Reply:
I think I found why it is undocumented:
* https://github.com/cypress-io/cypress-documentation/issues/1573
* https://github.com/cypress-io/cypress/pull/3929#issuecomment-482372968

Since `.all` is experimental, I would recommend looking into a different approach to this specific case.

Reply:
Here's another approach, less concise, but uses documented cypress features:
```
      let count = 0;
      cy.wait('@searchIndex').then((response) => {
        expect(response.response.body[0]).to.equal('filtered');
        count++;
      });
      cy.get('@searchIndex').then(() => {
        expect(count).to.equal(1);
      });
```